Runtime/ActionRouter/ActionRouterGroup.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using Padoru.Diagnostics;
+
+namespace Padoru.Core.ActionRouter
+{
+    public class ActionRouterGroup
+    {
+        public string Id { get; }
+
+        private readonly List<Action<object>> subscribers = new();
+        
+        public ActionRouterGroup(string id)
+        {
+            Id = id;
+        }
+
+        public void Invoke(object actionObject)
+        {
+            for (var index = subscribers.Count - 1; index >= 0; index--)
```

Comment:
Why is the for loop backwards?

Reply:
Because it can happen that when an action is invoked, an unsubscription is made, affecting the collection of actions that are being executed (in fact, it happened to me in the example I did).

Reply:
Makes sense